mbuild/compound.py

```diff
@@ -991,7 +991,13 @@ def _to_topology(self, atom_list, chains=None, residues=None):
         for atom in atom_list:
             # Chains
             for parent in atom.ancestors():
-                if chains and parent.name in chains:
+                if chains and atom.name in chains:
```

Comment:
This also needs to happen for the residues below. A few extra test cases for both naming chains and residues will probably raise confidence here for the mdtraj conversions.